### ISSUE TITLE
`conan info . --json` writes more than just the JSON to stdout

### ISSUE THREAD

**Issue by CryZe** [conan-io/conan#5895]
When you use `conan info . --json` to get the conan information formatted as JSON it sometimes writes additional information such as version resolution information into the same stdout stream, thus rendering the output invalid JSON:

```json
Version ranges solved
    Version range '~1.2.3' required by 'conanfile.py (MyProject/1.3.4)' resolved to 'the-dependency/1.2.3@author/stable' in local cache

[{"build_id": null, ... }]
```

This only happens when you don't explicitly specify an output json file and instead let it pipe the output through stdout (presumably because the version resolution message simply always writes to stdout, while the json can be written to either a file or stdout).

`Conan version 1.18.4 on Ubuntu 18.04.3 LTS`

To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by czoido** [comment]
Hi @CryZe,
Thanks a lot for reporting. As we have previously discussed in #1895 and #4359 for the console output of `conan info` could be a little bit tricky, but we'll take this into account for future changes :)

**Comment by sztomi** [comment]
Since this is getting a bit of attention, may I plug my more general suggestion: https://github.com/conan-io/conan/issues/1895 

Since then, many commands added --json which **writes a file**. This is workable, but slightly more awkward to program around than just capturing stdout.

**Comment by memsharded** [comment]
All the output in Conan 2.0 is sent to stderr, except the valuable load sent to stdout. Closing this as fixed in 2.0
